rotkehlchen/db/upgrades/v36_v37.py

```diff
@@ -31,36 +36,58 @@ def _update_history_events_schema(write_cursor: 'DBCursor', conn: 'DBConnection'
     notes TEXT,
     type TEXT NOT NULL,
     subtype TEXT NOT NULL,
-    counterparty TEXT,
-    extra_data TEXT,
     FOREIGN KEY(asset) REFERENCES assets(identifier) ON UPDATE CASCADE,
     UNIQUE(event_identifier, sequence_index)
     );""")
     new_entries = []
     with conn.read_ctx() as read_cursor:
         read_cursor.execute('SELECT * from history_events')
-
         for entry in read_cursor:
+            location = Location.deserialize_from_db(entry[4])
+            event_identifier = entry[1]
+            if location == Location.KRAKEN or event_identifier.startswith(b'rotki_events'):  # This is the rule at 1.27.1   # noqa: E501
+                entry_type = 0  # Pure history base entry
+            else:
+                entry_type = 1  # An evm event
             if entry[11] is None:
-                new_entries.append(entry[:11] + ('none',) + entry[12:])
+                new_entries.append([entry_type, *entry[:11], 'none'])  # turn NULL values to text `none`  # noqa: E501
             else:
-                new_entries.append(entry)
+                new_entries.append([entry_type, *entry[:12]])  # Don't change NON-NULL values  # noqa: E501
 
-    write_cursor.executemany('INSERT INTO history_events_copy VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ? ,? ,? ,?)', new_entries)  # noqa: E501
+    write_cursor.executemany('INSERT INTO history_events_copy VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)', new_entries)  # noqa: E501
     write_cursor.execute('DROP TABLE history_events')
     write_cursor.execute('ALTER TABLE history_events_copy RENAME TO history_events')
     log.debug('Exit _update_history_events_schema')
 
 
+def _create_new_tables(write_cursor: 'DBCursor') -> None:
+    """Create new tables"""
+    log.debug('Enter _create_new_tables')
+    write_cursor.execute("""
+        CREATE TABLE IF NOT EXISTS evm_events_info(
+            identifier INTEGER PRIMARY KEY,
+            counterparty TEXT,
+            product TEXT,
+            address TEXT,
+            extra_data TEXT,
+            FOREIGN KEY(identifier) REFERENCES history_events(identifier) ON UPDATE CASCADE ON DELETE CASCADE
+        );
+    """)  # noqa: E501
```

Comment:
As mentioned here: https://github.com/rotki/rotki/pull/5726/files#r1138584425

The author needs to specify here that the data **are not** carried over and need to be redecoded. Otherwise every other reader will see a bug here.

Also what happens to custom events? Would they not need to be carried over? Custom events **must never** be lost.

Also when/where is the redecoding done? Just let it run periodically? wouldn't you need to also clear all decoded events too?